dpnp/backend/kernels/dpnp_krnl_logic.cpp

```diff
@@ -30,68 +30,89 @@
 #include "queue_sycl.hpp"
 
 
-template <typename _DataType, typename _ResultType>
+template <typename _DataType>
+class dpnp_all_c_kernel;
+
+template <typename _DataType>
 void dpnp_all_c(void* array1_in, void* result1, const size_t size)
 {
-    _DataType* array_in = reinterpret_cast<_DataType*>(array1_in);
-    _ResultType* result = reinterpret_cast<_ResultType*>(result1);
+    cl::sycl::event event;
+
+    const _DataType* array_in = reinterpret_cast<const _DataType*>(array1_in);
+    bool* result = reinterpret_cast<bool*>(result1);
 
-    bool res = true;
+    result[0] = true;
 
-    for (size_t i = 0; i < size; ++i)
+    if ((array1_in != nullptr) && (result1 != nullptr))
     {
-        if (!array_in[i])
-        {
-            res = false;
-            break;
-        }
-    }
+        cl::sycl::range<1> gws(size);
+        auto kernel_parallel_for_func = [=](cl::sycl::id<1> global_id) {
+            size_t i = global_id[0];
 
-    result[0] = res;
+            if (!array_in[i])
+            {
+                result[0] = false;
```

Comment:
Sorry but I think it will not work in parallel environment.
The problem is in accessing the same element from all threads. 
Let's imagine input array with 100 elements. This code emits 100 threads (gws=100). All threads will try to write their value `true` or `false` into the same memory address (result[0]).
Final result will strongly depend on the order of threads execution. Strictly speaking, the result will be overwritten by last active thread.

Are you sure this algorithm should return only one value for all number of input values? I didn't understand this by reading numpy documentation. They mentioned scalar and array as output at the same time.

Reply:
No, I'm wrong. It will work. Dismiss my previous comment.